mobilecoind-json/src/data_types.rs

```diff
@@ -491,21 +491,63 @@ pub struct JsonMaskedAmount {
     pub commitment: String,
     pub masked_value: JsonU64,
     pub masked_token_id: String,
+    pub version: Option<u32>,
 }
 
-impl From<&MaskedAmount> for JsonMaskedAmount {
-    fn from(src: &MaskedAmount) -> Self {
-        Self {
-            commitment: hex::encode(src.get_commitment().get_data()),
-            masked_value: JsonU64(src.get_masked_value()),
-            masked_token_id: hex::encode(src.get_masked_token_id()),
+impl From<&mc_api::external::TxOut_oneof_masked_amount> for JsonMaskedAmount {
+    fn from(src: &mc_api::external::TxOut_oneof_masked_amount) -> Self {
+        match src {
+            mc_api::external::TxOut_oneof_masked_amount::v1(src) => Self {
+                commitment: hex::encode(src.get_commitment().get_data()),
+                masked_value: JsonU64(src.get_masked_value()),
+                masked_token_id: hex::encode(src.get_masked_token_id()),
+                version: None,
+            },
+            mc_api::external::TxOut_oneof_masked_amount::v2(src) => Self {
+                commitment: hex::encode(src.get_commitment().get_data()),
+                masked_value: JsonU64(src.get_masked_value()),
+                masked_token_id: hex::encode(src.get_masked_token_id()),
+                version: Some(2),
+            },
+        }
```

Comment:
I tried this, but I reverted it because it doesn't actually build

I think the way it was is the best way it can actually be, the src is shadowed by what is matched in the match expression. We could try adding more helpers to this externally generated proto structure to make the code you wrote work, but I don't think it's really worth it for now